pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -62,6 +62,7 @@ func ProvideService(
 				log:                            logger,
 				kvstore:                        namespacedKVStore,
 				backwardsCompatibilityDisabled: features.IsEnabled(featuremgmt.FlagDisableSecretsCompatibility),
+				fallback:                       store,
```

Comment:
```suggestion
				// as the plugin is installed, secretsKVStoreSQL is now replaced with an instance of secretsKVStorePlugin with the sql store as a fallback (used for migration and in case a secret is not found). 
				fallback:                       store,
```

This part took me a while to follow correctly, I am just suggesting adding a similar comment for the future